### PR TITLE
[IOTDB-2392] Memory control of raft log in cluster 

### PR DESCRIPTION
**Pull request by MrQuansy**
See jira: https://issues.apache.org/jira/projects/IOTDB/issues/IOTDB-2392.

### CODE REVIEW

**Review by jt2594838 — APPROVED** [review]
Good job. But the rejection is based on the number of entries instead of their memory footprint. Could you please add some todos that we use precise memory calculation for rejection?

File: cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java [hunk 1]
```diff
@@ -2034,6 +2073,10 @@ private long appendEntries(
     }
 
     synchronized (logManager) {
+      if ((logManager.getCommitLogIndex() - logManager.getMaxHaveAppliedCommitIndex())
+          > config.getUnAppliedRaftLogNumForRejectThreshold()) {
+        return Response.RESPONSE_TOO_BUSY;
+      }
```
Comment (jt2594838) [inline cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java:2079]: Adding some wait time here may improve robustness.

Comment (MrQuansy) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java]: Ok, I add some statements to wait some time before rejection.

**Review by jt2594838 — APPROVED** [review]

**Review by LebronAl — APPROVED** [review]
LGTM

**Pull request merged**